src/test/java/roomescape/service/JwtTokenProviderTest.java

```diff
@@ -0,0 +1,47 @@
+package roomescape.service;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import static roomescape.model.Role.MEMBER;
+
+import org.assertj.core.api.SoftAssertions;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+
+import io.jsonwebtoken.Claims;
+import roomescape.model.User;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.DEFINED_PORT)
+class JwtTokenProviderTest {
+
+    @Autowired
+    private JwtTokenProvider jwtTokenProvider;
+
+    @DisplayName("주어진 사용자로 토큰을 생성한다.")
+    @Test
+    void should_create_token_when_given_user() {
+        User user = new User(1L, "썬", MEMBER, "sun@email.com", "1234");
+
+        String token = jwtTokenProvider.createToken(user);
+
+        assertThat(token).isNotBlank();
+    }
+
+    @DisplayName("주어진 토큰으로 payload를 반환한다.")
+    @Test
+    void should_get_payload_when_given_token() {
+        User user = new User(1L, "썬", MEMBER, "sun@email.com", "1234");
+        String token = jwtTokenProvider.createToken(user);
+
+        Claims claims = jwtTokenProvider.getPayload(token);
+
+        String userId = claims.get("user_id", String.class);
+        String role = claims.get("role", String.class);
+        SoftAssertions.assertSoftly(softAssertions -> {
+            softAssertions.assertThat(userId).isEqualTo("1");
+            softAssertions.assertThat(role).isEqualTo("MEMBER");
+        });
```

Comment:
```suggestion
        assertSoftly(softAssertions -> {
            softAssertions.assertThat(userId).isEqualTo("1");
            softAssertions.assertThat(role).isEqualTo("MEMBER");
        });
```

Q3) 크게 문제가 되는건지 잘 모르겠습니다. create를 통해 발급이 이뤄지는지도 테스트하고 있고, 테스트가 실패하면 제대로 동작하지 않는걸 확인하는 것 아닌가요? 객체에 단위 테스트를 진행할 때에도, 객체를 먼저 만들고 해당 객체가 수행하는 기능을 테스트하지 않나요? 🤔
만약 정상적인 토큰이 생성되는지를 테스트하고 싶다면 라이브러리에 대한 도전인데, 라이브러리를 믿지 않는다면 전부 하나 하나 개발하고 테스트해야할 것 같아요.

Reply:
> 만약 정상적인 토큰이 생성되는지를 테스트하고 싶다면 라이브러리에 대한 도전인데, 

답변 감사합니다! 어디까지 확신해야 할지 모르겠다는 생각에 시작했던 고민인 것 같네요😅 리뷰어님 말씀을 들으니 믿고 가는 것이 맞는 것 같아요 :)